*** electronic_assembly_interconnect_and_packaging_design_arm.lfepm_31_5000100006.exp	2012-11-17 11:50:18.000000000 -0600
--- patched_electronic_assembly_interconnect_and_packaging_design_arm.lfepm_31_5000100006.exp	2012-11-17 12:04:24.000000000 -0600
***************
*** 17852,17858 ****
        leaf_node : PRODUCT_VIEW_DEFINITION_OR_REFERENCE := SELF\Product_occurrence_definition_relationship.related_view;  -- RELATED_VIEW : This ATTRIBUTE was RENAMED in the EXPRESS E2 source, this is the new name implemented as DERIVE
        root_node : PRODUCT_VIEW_DEFINITION := SELF\Product_occurrence_definition_relationship.relating_view;  -- RELATING_VIEW : This ATTRIBUTE was RENAMED in the EXPRESS E2 source, this is the new name implemented as DERIVE
  
!     : product_view_definition_or_reference := location[HIINDEX(location)]\Product_occurrence_definition_relationship.related_view;
  UNIQUE
    UR1 : location;
  END_ENTITY;
--- 17852,17858 ----
        leaf_node : PRODUCT_VIEW_DEFINITION_OR_REFERENCE := SELF\Product_occurrence_definition_relationship.related_view;  -- RELATED_VIEW : This ATTRIBUTE was RENAMED in the EXPRESS E2 source, this is the new name implemented as DERIVE
        root_node : PRODUCT_VIEW_DEFINITION := SELF\Product_occurrence_definition_relationship.relating_view;  -- RELATING_VIEW : This ATTRIBUTE was RENAMED in the EXPRESS E2 source, this is the new name implemented as DERIVE
  
! --bug #4667    : product_view_definition_or_reference := location[HIINDEX(location)]\Product_occurrence_definition_relationship.related_view;
  UNIQUE
    UR1 : location;
  END_ENTITY;
***************
*** 30264,30276 ****
  (* Implicit interfaced from: Assembly_structure_arm *)
  (* E2: SUBTYPE_CONSTRAINT assembly_component_relationship_constraint FOR ASSEMBLY_COMPONENT_RELATIONSHIP; *)
  -- generated from SUBTYPE_CONSTRAINT assembly_component_relationship_constraint
! 
  RULE total_over_assembly_component_relationship_constraint FOR( ASSEMBLY_COMPONENT_RELATIONSHIP);
  WHERE
!   wr1 : SIZEOF(QUERY(p <* ASSEMBLY_COMPONENT_RELATIONSHIP | SIZEOF(['Ap210_electronic_assembly_interconnect_and_packaging_design_arm.COMPONENT_UPPER_LEVEL_IDENTIFICATION'] * TYPEOF(p) ) = 0)) = 0;
!   wr2 : SIZEOF(QUERY(p <* ASSEMBLY_COMPONENT_RELATIONSHIP | SIZEOF(['Ap210_electronic_assembly_interconnect_and_packaging_design_arm.MULTI_LEVEL_REFERENCE_DESIGNATOR'] * TYPEOF(p) ) = 0)) = 0;
!   wr3 : SIZEOF(QUERY(p <* ASSEMBLY_COMPONENT_RELATIONSHIP | SIZEOF(['Ap210_electronic_assembly_interconnect_and_packaging_design_arm.NEXT_ASSEMBLY_USAGE'] * TYPEOF(p) ) = 0)) = 0;
!   wr4 : SIZEOF(QUERY(p <* ASSEMBLY_COMPONENT_RELATIONSHIP | SIZEOF(['Ap210_electronic_assembly_interconnect_and_packaging_design_arm.PROMISSORY_USAGE'] * TYPEOF(p) ) = 0)) = 0;
  END_RULE;
  
  (* Implicit interfaced from: Appearance_assignment_arm *)
--- 30264,30277 ----
  (* Implicit interfaced from: Assembly_structure_arm *)
  (* E2: SUBTYPE_CONSTRAINT assembly_component_relationship_constraint FOR ASSEMBLY_COMPONENT_RELATIONSHIP; *)
  -- generated from SUBTYPE_CONSTRAINT assembly_component_relationship_constraint
! --bug #4662
  RULE total_over_assembly_component_relationship_constraint FOR( ASSEMBLY_COMPONENT_RELATIONSHIP);
  WHERE
!   wr1 : SIZEOF(QUERY(p <* ASSEMBLY_COMPONENT_RELATIONSHIP | 
!   SIZEOF(['AP210_ELECTRONIC_ASSEMBLY_INTERCONNECT_AND_PACKAGING_DESIGN_ARM.COMPONENT_UPPER_LEVEL_IDENTIFICATION',
!           'AP210_ELECTRONIC_ASSEMBLY_INTERCONNECT_AND_PACKAGING_DESIGN_ARM.MULTI_LEVEL_REFERENCE_DESIGNATOR',
!           'AP210_ELECTRONIC_ASSEMBLY_INTERCONNECT_AND_PACKAGING_DESIGN_ARM.NEXT_ASSEMBLY_USAGE',
!           'AP210_ELECTRONIC_ASSEMBLY_INTERCONNECT_AND_PACKAGING_DESIGN_ARM.PROMISSORY_USAGE'] * TYPEOF(p) ) = 0)) = 0;
  END_RULE;
  
  (* Implicit interfaced from: Appearance_assignment_arm *)
